### PR TITLE
Fix emit for `export {...}` with ES6 target and non-ES6 modules

### LINKED ISSUE #5175: `export {...}` doesn't work with ES6 target and non-ES6 modules
In #4811 we enabled generating code for ES6 with non-ES6 modules (which, for example, is useful when targeting Node.js v4). This work didn't properly handle `export {...}` declarations.

Compile the following with `-t es6 -m commonjs`:

```typescript
export function foo() { }
function bar() { }
export { bar };
```

Expected output is:

```javascript
function foo() { }
exports.foo = foo;
function bar() { }
exports.bar = bar;
```

Actual output is:

```javascript
function foo() { }
exports.foo = foo;
function bar() { }
```

I will be putting up a fix for the problem shortly.

### PR DESCRIPTION
**Pull request by ahejlsberg**
Fixes #5175.

### CODE REVIEW

**Comment by ahejlsberg** [comment]
@mhegazy @weswigham You may want to look this over.

File: tests/baselines/reference/exportsAndImports1-es6.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+module M {
+    export var x;
+}
+module N {
+    export interface I {
+    }
+}
+type T = number;
+import a = M.x;
+
```
Comment (weswigham) [inline tests/baselines/reference/exportsAndImports1-es6.js]: Add in a reassignment to `v` (ie, `v = 10`) here. Observe that we emit `v = 10` but no corresponding `exports.v = v`. I'm not sure lifting all the exports from where the export statement is to wherever the variable's declaration is is correct for this export form.

**Comment by mhegazy** [comment]
:+1:

File: tests/baselines/reference/exportsAndImports1-es6.js (hunk 1 above)
Comment (ahejlsberg) [inline tests/baselines/reference/exportsAndImports1-es6.js]: That's an orthogonal issue already being discussed in #4101.

Comment (weswigham) [inline tests/baselines/reference/exportsAndImports1-es6.js]: Ahh, okay, didn't know about the old issue.

**Comment by weswigham** [comment]
:+1:

**Pull request merged**